https://www.kaggle.com/code/radek1/eda-a-look-at-the-data-training-splits

https://www.kaggle.com/competitions/otto-recommender-system/discussion/363843

In [1]:
import pandas as pd
import numpy as np
import pickle

dataPath = "otto-recommender-system"

In [9]:
df_submit_sample = pd.read_csv("otto-recommender-system/sample_submission.csv")
df_submit_sample

session_type                labels
0        12899779_clicks  129004 126836 118524
1         12899779_carts  129004 126836 118524
2        12899779_orders  129004 126836 118524
3        12899780_clicks  129004 126836 118524
4         12899780_carts  129004 126836 118524
...                  ...                   ...
5015404   14571580_carts  129004 126836 118524
5015405  14571580_orders  129004 126836 118524
5015406  14571581_clicks  129004 126836 118524
5015407   14571581_carts  129004 126836 118524
5015408  14571581_orders  129004 126836 118524

[5015409 rows x 2 columns]

In [2]:
df_test = pd.read_parquet(f"{dataPath}/test.parquet")
df_test

session      aid          ts  type
0        12899779    59625  1661724000     0
1        12899780  1142000  1661724000     0
2        12899780   582732  1661724058     0
3        12899780   973453  1661724109     0
4        12899780   736515  1661724136     0
...           ...      ...         ...   ...
6928118  14571577  1141710  1662328774     0
6928119  14571578   519105  1662328775     0
6928120  14571579   739876  1662328775     0
6928121  14571580   202353  1662328781     0
6928122  14571581  1100210  1662328791     0

[6928123 rows x 4 columns]

In [3]:
df_train = pd.read_parquet(f"{dataPath}/train.parquet")
df_train

session      aid          ts  type
0                 0  1517085  1659304800     0
1                 0  1563459  1659304904     0
2                 0  1309446  1659367439     0
3                 0    16246  1659367719     0
4                 0  1781822  1659367871     0
...             ...      ...         ...   ...
216716091  12899776  1737908  1661723987     0
216716092  12899777   384045  1661723976     0
216716093  12899777   384045  1661723986     0
216716094  12899778   561560  1661723983     0
216716095  12899778    32070  1661723994     0

[216716096 rows x 4 columns]

In [4]:
with open(f"{dataPath}/id2type.pkl", "rb") as fh:
    id2type = pickle.load(fh)
with open(f"{dataPath}/type2id.pkl", "rb") as fh:
    type2id = pickle.load(fh)
id2type, type2id

(['clicks', 'carts', 'orders'], {'clicks': 0, 'carts': 1, 'orders': 2})

In [5]:
df_train.shape, df_test.shape

((216716096, 4), (6928123, 4))

The train dataset contains 216716096 datapoints with test containing only 6928123.

Proportion of test to train:

In [6]:
df_test.shape[0]/df_train.shape[0]

0.03196865912534711

The size of the test set is ~3.1% of the train set. This can give us an idea of how lightweight the inference is likely to be compared to training.

How many session are there in train and test?

In [7]:
df_train.session.unique().shape[0], df_test.session.unique().shape[0]

(12899779, 1671803)

In [8]:
df_test.session.unique().shape[0]/df_train.session.unique().shape[0]

0.12959935205091497